src/main/java/List.java

```diff
@@ -0,0 +1,221 @@
+        numItems += 1;
+    }
+
+    private void outputAddedMessage(Task task) {
+        System.out.println("\tGot it. I've added this task:");
+        System.out.println("\t  " + task);
+        System.out.println("\tNow you have " + (numItems+1) + " tasks in the list.");
+    }
+
+    private String extractTodoDescription(String line) {
+        String todoDescription;
+        todoDescription = line.trim().replace("todo ", "");
+
+        return todoDescription;
+    }
+
+    private String extractDeadlineDescription(String line) {
+        String deadlineDescription;
+        final int indexOfDeadlinePrefix = line.indexOf("/by");
+        deadlineDescription = line.substring(0, indexOfDeadlinePrefix).trim().replace("deadline ", "");
+
+        return deadlineDescription;
+    }
+
+    private String extractDeadlineDate(String line) {
+        String deadlineDate;
+        final int indexOfDeadlinePrefix = line.indexOf("/by");
+        deadlineDate = line.substring(indexOfDeadlinePrefix).trim().replace("/by ", "");
+
+        return deadlineDate;
+    }
+
+    private String extractEventDescription(String line) {
+        String eventDescription;
+        final int indexOfStartDatePrefix = line.indexOf("/from");
+        final int indexOfEndDatePrefix = line.indexOf("/to");
+        if (indexOfEndDatePrefix > indexOfStartDatePrefix) {
+            eventDescription = line.substring(0, indexOfStartDatePrefix).trim().replace("event ", "");
+        } else {
+            eventDescription = line.substring(0, indexOfEndDatePrefix).trim().replace("event ", "");
+        }
+
+        return eventDescription;
+    }
+
+    private String extractEventEndDate(String line) {
+        String eventEndDate;
+        final int indexOfStartDatePrefix = line.indexOf("/from");
+        final int indexOfEndDatePrefix = line.indexOf("/to");
+        if (indexOfEndDatePrefix > indexOfStartDatePrefix) {
+            eventEndDate = line.substring(indexOfEndDatePrefix).trim().replace("/to ", "");
+        } else {
+            eventEndDate = line.substring(indexOfEndDatePrefix, indexOfStartDatePrefix).trim().replace("/to ", "");
+        }
+
+        return eventEndDate;
+    }
+
+    private String extractEventStartDate(String line) {
+        String eventStartDate;
+        final int indexOfStartDatePrefix = line.indexOf("/from");
+        final int indexOfEndDatePrefix = line.indexOf("/to");
+        if (indexOfStartDatePrefix > indexOfEndDatePrefix) {
+            eventStartDate = line.substring(indexOfStartDatePrefix).trim().replace("/from ", "");
+        } else {
+            eventStartDate = line.substring(indexOfStartDatePrefix, indexOfEndDatePrefix).trim().replace("/from ", "");
+        }
+
+        return eventStartDate;
+    }
+
+    public void printList() {
+        System.out.println("\tHere are the tasks in your list:");
+        for (int i = 0; i < numItems; i++) {
+            System.out.println("\t" + (i + 1) + "." + itemList[i]);
```

Comment:
I like the use of the toString() override to allow easier printing of the elements in itemList<Task>.

Reply:
Thank you for your suggestion, the toString() function is already being called in itemList[i].